addons/account_payment/controllers/payment.py

```diff
@@ -30,11 +30,15 @@ def invoice_transaction(self, invoice_id, access_token, **kwargs):
         except AccessError:
             raise ValidationError(_("The access token is invalid."))
 
-        kwargs['reference_prefix'] = None  # Allow the reference to be computed based on the invoice
+        self._check_illegal_kwargs(kwargs)
         logged_in = not request.env.user._is_public()
         partner = request.env.user.partner_id if logged_in else invoice_sudo.partner_id
-        kwargs['partner_id'] = partner.id
-        kwargs.pop('custom_create_values', None)  # Don't allow passing arbitrary create values
+        kwargs.update({
+            'reference_prefix': None,  # Allow the reference to be computed based on the order
+            'partner_id': partner.id, # Prevent passing an arbitrary partner by forcing it.
```

Comment:
Isn't the partner_id one of the kwargs that were checked? You can't pass a partner_id anyway, so you don't force it, you add it here. That's different, this comment isn't correct here IMO